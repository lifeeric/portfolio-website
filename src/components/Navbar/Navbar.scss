@import "../../components/Theme/variables";

.navbar {
  // Toggle
  &__toggle {
    display: none;
  }

  &__list {
    margin: 0;
    padding: 0;
    overflow: hidden;
    list-style: none;
  }

  &__item {
    display: inline-block;
    margin-right: 30px;
    font-family: montserrat, sans-serif;
    font-weight: 500;
    cursor: pointer;
    color: $blackColor;
    transition: 1s ease;
    border-bottom: 3px solid transparent;

    // hover
    &:hover {
      color: $borderActive;
    }

    // last item
    &:last-child {
      margin-right: 0;
    }
  }

  &__item > .active {
    border-bottom: 3px solid $borderActive;
  }
}

// media

@media (max-width: 600px) {
  // Menu Icon

  .navbar {
    width: 100%;
    height: 100%;
    position: fixed;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    transform: translateX(-100%);
    z-index: 100;

    // when clicked on the button
    &.showMenu {
      transform: translateX(0);
    }

    // close icon
    &__close {
      width: 35px;
      height: 5px;
      position: absolute;
      top: 30px;
      right: 25px;
      z-index: 20;
      cursor: pointer;

      &:before,
      &:after {
        content: "";
        position: absolute;
        top: 0;
        right: 0;
        width: 100%;
        height: 100%;
        background: $whiteColor;
        transform: rotate(45deg);
      }

      &:after {
        transform: rotate(-45deg);
      }
    }

    // ul
    &__list {
      margin: 50% auto;
      width: 100%;
      text-align: center;
    }
    // item
    &__item {
      display: block !important;
      font-size: 24px;
      margin-top: 15px;
      margin-right: 0;
      color: $whiteColor;

      &--active {
        border: none;
        color: $borderActive;
      }
    }
  }

  // Menu clip path
  .navbar {
    transition: clip-path 0.8s cubic-bezier(0.645, 0.045, 0.355, 1) 0s,
      transform 0s cubic-bezier(0.645, 0.045, 0.355, 1) 0.8s;
    background-color: #141414;
    clip-path: circle(0% at 102% -12px);

    &.showMenu {
      transition: clip-path 0.8s cubic-bezier(0.645, 0.045, 0.355, 1) 0s,
        transform 0s cubic-bezier(0.645, 0.045, 0.355, 1) 0s;
      clip-path: circle(72%);
    }
  }

  .toggle {
    cursor: pointer;
    &__right {
      position: relative;
    }

    &__dots {
      position: absolute;
      top: 30px;
      right: 25px;
    }

    // Dots
    @for $var from 1 through 3 {
      &__dot--#{$var} {
        width: 33px - ($var * 3);
        height: 3px;
        background: $borderActive;
        margin-top: 3px;
        border-radius: 5px;
        border-collapse: separate;
      }
    }
  }
}
