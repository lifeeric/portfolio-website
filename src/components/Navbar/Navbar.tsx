import * as React from "react"
import { useState } from "react"
import { NavLinkItem } from "./NavLinkItem/NavLinkItem"

import "./Navbar.scss"

const links = [
  {
    to: "home",
    offset: -120,
    name: "Me",
  },
  {
    to: "project",
    offset: -70,
    name: "Projects",
  },
  {
    to: "skill",
    offset: -70,
    name: "Skills",
  },
  {
    to: "contact",
    offset: -70,
    name: "Contact",
  },
]

export const Navbar: React.FC = () => {
  const [toggle, setToggle] = useState<boolean>(false)

  // Menu Toggler
  const menuToggleHandler = (): void => {
    setToggle(toggle => !toggle)
  }

  const closeToggleHandler = (): void => {
    setToggle(false)
    console.log(toggle)
  }

  return (
    <>
      <div className="toggle">
        <div className="toggle_right" onClick={menuToggleHandler}>
          <div className="toggle__dots">
            <div className="toggle__dot--1"></div>
            <div className="toggle__dot--2"></div>
            <div className="toggle__dot--3"></div>
          </div>
        </div>
      </div>

      {/* Navbar */}
      <nav className={`navbar ${toggle && `showMenu`}`}>
        <span className="navbar__close" onClick={menuToggleHandler}></span>
        <ul className="navbar__list">
          {links.map(link => (
            <NavLinkItem key={link.to} {...link} onClick={closeToggleHandler} />
          ))}
        </ul>
      </nav>
    </>
  )
}
