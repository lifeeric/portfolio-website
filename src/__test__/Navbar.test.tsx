import React from "react"
import { shallow } from "enzyme"

// Import Components
import { Navbar } from "../components/Navbar/Navbar"

// Start writing test
describe("First Test", () => {
  it("should render two <NavigationItem /> element if not authenicated", () => {
    const component = shallow(<Navbar />)
    expect(component).toMatchSnapshot()
  })
})
