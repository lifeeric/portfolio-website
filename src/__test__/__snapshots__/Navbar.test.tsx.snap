// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First Test should render two <NavigationItem /> element if not authenicated 1`] = `
<Fragment>
  <div
    className="toggle"
  >
    <div
      className="toggle_right"
      onClick={[Function]}
    >
      <div
        className="toggle__dots"
      >
        <div
          className="toggle__dot--1"
        />
        <div
          className="toggle__dot--2"
        />
        <div
          className="toggle__dot--3"
        />
      </div>
    </div>
  </div>
  <nav
    className="navbar false"
  >
    <span
      className="navbar__close"
      onClick={[Function]}
    />
    <ul
      className="navbar__list"
    >
      <Component
        key="home"
        name="Me"
        offset={-120}
        onClick={[Function]}
        to="home"
      />
      <Component
        key="project"
        name="Projects"
        offset={-70}
        onClick={[Function]}
        to="project"
      />
      <Component
        key="skill"
        name="Skills"
        offset={-70}
        onClick={[Function]}
        to="skill"
      />
      <Component
        key="contact"
        name="Contact"
        offset={-70}
        onClick={[Function]}
        to="contact"
      />
    </ul>
  </nav>
</Fragment>
`;
